internal/telemetry/tracker_test.go

```diff
@@ -0,0 +1,49 @@
+package telemetry
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestServiceConstructor(t *testing.T) {
+	assert.RegisterOpts(reflect.TypeOf(&service{}), cmp.AllowUnexported(service{}))
+
+	t.Run(fmt.Sprintf("Should create the expected Service"), func(t *testing.T) {
+		constructedService := NewService(ModeOn, "userID", "executionID", "command")
+		expectedService := &service{
+			userID:      "userID",
+			executionID: "executionID",
+			command:     "command",
+			tracker:     &segmentTracker{},
+		}
+		assert.Equal(t, expectedService, constructedService)
+	})
+}
+
+func TestEventTracking(t *testing.T) {
+
+	t.Run(fmt.Sprintf("Should track the expected event"), func(t *testing.T) {
+		trackerService := NewService(modeTest, "userID", "executionID", "command")
+		trackerService.TrackEvent(EventTypeCommandError, EventData{Key: EventDataKeyErr, Value: fmt.Errorf("error")})
+		trackerService.TrackEvent(EventTypeCommandError, EventData{Key: EventDataKeyErr, Value: fmt.Errorf("error")})
```

Comment:
are the two `TrackEvent` calls here both required?

--

the linter's going to complain about `fmt.Errorf` here, switch it to `errors.New` instead (for L#39 below too)